spec.html

```diff
@@ -36261,6 +36277,21 @@ <h1>Properties of the Array Prototype Object</h1>
         <p>The Array prototype object is specified to be an Array exotic object to ensure compatibility with ECMAScript code that was created prior to the ECMAScript 2015 specification.</p>
       </emu-note>
 
+      <emu-clause id="sec-array.prototype.at">
+        <h1>Array.prototype.at ( _index_ )</h1>
+        <emu-alg>
+          1. Let _O_ be ? ToObject(*this* value).
+          1. Let _len_ be ? LengthOfArrayLike(_O_).
+          1. Let _relativeIndex_ be ? ToIntegerOrInfinity(_index_).
+          1. If _relativeIndex_ &ge; 0, then
+            1. Let _k_ be _relativeIndex_.
+          1. Else,
+            1. Let _k_ be _len_ + _relativeIndex_.
```

Comment:
This can end up performing arithmetic on an infinity (specifically negative infinity), which we generally avoid so we don't have to define it. Can we guard against it here (and in the other methods), for example by adding

```
1. If _relativeIndex_ is not finite, return *undefined*.
```
after the `ToIntegerOrInfinity` step?

(Or we could define in [the relevant section](https://tc39.es/ecma262/multipage/notational-conventions.html#sec-mathematical-operations) that the sum of an infinity and any finite value is the infinity, I guess.)

Reply:
I kind of prefer just permitting this arithmetic and not having to remember to avoid it everywhere.

Reply:
I'm fine with adding the step, but I do wonder why not define it. What are the concrete risks?

Reply:
Defining it seems fine, if we want to do that. I can make a PR or you can include it in this one.

Reply:
I've added a sentence that refers to the usual definition of arithmetic on extended reals. We could be more surgical here and only define e.g. + and - between a finite MV and an infinity value, if you think this is too handwavy, but I didn't see the need to explicitly give the definitions since we don't try to be formal with math underpinnings to begin with.

Reply:
> When applied to extended mathematical values, the operators refer to the usual mathematical operations over the extended real numbers;

The spec doesn't define what "extended real numbers" are.

> indeterminate forms are undefined and their use in this specification should be considered an editorial error.

(You should maybe change "undefined" to "not defined", so that nobody thinks you mean `*undefined*`.)

Personally, I think I'd prefer to be told which operations involving infinities are allowed in the spec.

Reply:
> The spec doesn't define what "extended real numbers" are.

True, but it doesn't define "real numbers" either.

Reply:
so, to confirm, this means `-Infinity + finiteNumber === -Infinity`?

Reply:
Yeah, I think here's where we need to make an editorial decision for the level of mathematical background we can assume. We can, for example, reproduce these definitions [from Wikipedia](https://en.wikipedia.org/wiki/Extended_real_number_line#Arithmetic_operations) in the spec, along with definitions of &ge; etc. But I'm not convinced that level of formality is needed. I think it's reasonable to reference a searchable mathematical phrase, like "extended real numbers" to make the information easily searchable for unfamiliar folks.

Reply:
> so, to confirm, this means -Infinity + finiteNumber === -Infinity?

Yeah

Reply:
I think a Note with an explicit link (to Wikipedia, or whatever) would help.

Reply:
I'll bring this up for discussion at the next editor call.